packages/runtime/src/interpreter/opcode.ts

```diff
@@ -6,85 +6,87 @@ import type { TEALStack } from "../types";
+  checkIndexBound (idx: number, arr: any[], line: number): void {
     if (!(idx >= 0 && idx < arr.length)) {
-      throw new TealError(ERRORS.TEAL.INDEX_OUT_OF_BOUND);
+      throw new TealError(ERRORS.TEAL.INDEX_OUT_OF_BOUND, { line: line });
     }
   }
 
-  assertArrLength (arr: Uint8Array[] | BigInt[]): void {
+  assertArrLength (arr: Uint8Array[] | BigInt[], line: number): void {
     if (!arr.length || arr.length > MAX_UINT8 + 1) {
-      throw new TealError(ERRORS.TEAL.ASSERT_ARR_LENGTH);
+      throw new TealError(ERRORS.TEAL.ASSERT_ARR_LENGTH, { line: line });
     }
   }
 
-  assertBigInt (a: unknown): bigint {
+  assertBigInt (a: unknown, line: number): bigint {
     if (typeof a === "undefined" || typeof a !== "bigint") {
       throw new TealError(ERRORS.TEAL.INVALID_TYPE, {
         expected: "uint64",
-        actual: typeof a
+        actual: typeof a,
+        line: line
       });
     }
     return a;
   }
 
-  assertBytes (b: unknown): Uint8Array {
+  assertBytes (b: unknown, line: number): Uint8Array {
     if (typeof b === 'undefined' || !(b instanceof Uint8Array)) {
       throw new TealError(ERRORS.TEAL.INVALID_TYPE, {
         expected: "byte[]",
-        actual: typeof b
+        actual: typeof b,
+        line: line
       });
     }
     return b;
   }
 
-  assertUint8 (a: bigint): bigint {
+  assertUint8 (a: bigint, line: number): bigint {
     if (a < MIN_UINT8 || a > MAX_UINT8) {
-      throw new TealError(ERRORS.TEAL.INVALID_UINT8);
+      throw new TealError(ERRORS.TEAL.INVALID_UINT8, { line: line });
     }
     return a;
   }
 
-  subString (start: bigint, end: bigint, byteString: Uint8Array): Uint8Array {
+  subString (start: bigint, end: bigint, byteString: Uint8Array, line: number): Uint8Array {
     if (end < start) {
-      throw new TealError(ERRORS.TEAL.SUBSTRING_END_BEFORE_START);
+      throw new TealError(ERRORS.TEAL.SUBSTRING_END_BEFORE_START, { line: line });
     }
     if (start > byteString.length || end > byteString.length) {
-      throw new TealError(ERRORS.TEAL.SUBSTRING_RANGE_BEYOND);
+      throw new TealError(ERRORS.TEAL.SUBSTRING_RANGE_BEYOND, { line: line });
     }
 
     return byteString.slice(Number(start), Number(end));
   }
 
   // assert if known transaction field is passed
-  assertTxFieldDefined (str: string): void {
+  assertTxFieldDefined (str: string, line: number): void {
     if (TxnFields[str] === undefined) {
-      throw new TealError(ERRORS.TEAL.UNKOWN_TRANSACTION_FIELD, { field: str });
+      throw new TealError(ERRORS.TEAL.UNKOWN_TRANSACTION_FIELD, { field: str, line: line });
     }
   }
 
   // assert if known global field is passed
-  assertGlobalDefined (str: string): void {
+  assertGlobalDefined (str: string, line: number): void {
     if (GlobalFields[str] === undefined) {
-      throw new TealError(ERRORS.TEAL.UNKOWN_GLOBAL_FIELD, { field: str });
+      throw new TealError(ERRORS.TEAL.UNKOWN_GLOBAL_FIELD, { field: str, line: line });
     }
   }
```

Comment:
Please add a doc string (maybe at the top of file) explaining `line`

Reply:
yes